src/index.js

```diff
@@ -18,21 +18,45 @@ const TCP = and(IP, base('tcp'))
 const UDP = and(IP, base('udp'))
 const UTP = and(UDP, base('utp'))
 
+const TCP_IPFS = or(
```

Comment:
`mafmt.IPFS(ma)` would match TCP, WebRTC and WS, even if I do something like 

```
!mafmt.WS(ma) && !mafmt.<some other transport/proto>(ma) && mafmt.IPFS(ma)
```

`mafmt.IPFS(ma)` would still match them, since it matches everything that has `/ipfs` in it. I think a better way would be to allow mafmt.TCP(), mafmt.WS(), etc... to also match `/ipfs` fragments. The reason I did it in this way is to avoid breaking some other functionality, but I think I'm being overly cautious, don't think anything else would break.